src/ophyd_async/core/signal.py

```diff
@@ -272,6 +284,91 @@ def soft_signal_r_and_backend(
     return (signal, backend)
 
 
+async def assert_value(signal: SignalR[T], value: Any) -> None:
+    """Assert a signal's value and compare it an expected signal.
+
+    Parameters
+    ----------
+    signal:
+        signal with get_value.
+    value:
+        The expected value from the signal.
+
+    Notes
+    -----
+    Example usage::
+        await assert_value(signal, value)
+
+    """
+    assert await signal.get_value() == value
+
+
+async def assert_reading(
+    readable: AsyncReadable, reading: Mapping[str, Reading]
+) -> None:
+    """Assert readings from readable.
+
+    Parameters
+    ----------
+    readable:
+        Callable with readable.read function that generate readings.
+
+    reading:
+        The expected readings from the readable.
+
+    Notes
+    -----
+    Example usage::
+        await assert_reading(readable, reading)
+
+    """
+    assert await readable.read() == reading
+
+
+async def assert_configuration(
+    configurable: AsyncConfigurable,
+    configuration: Mapping[str, Reading],
+) -> None:
+    """Assert readings from Configurable.
+
+    Parameters
+    ----------
+    configurable:
+        Configurable with Configurable.read function that generate readings.
+
+    configuration:
+        The expected readings from configurable.
+
+    Notes
+    -----
+    Example usage::
+        await assert_configuration(configurable configuration)
+
+    """
+    assert await configurable.read_configuration() == configuration
+
+
+def assert_emitted(docs: Mapping[str, list[DocumentType]], **numbers: int):
```

Comment:
Good point, @coretl do you think something like this would work?
 
```python
from typing import Union
from event_model import ALL_DOCUMENTS

def assert_emitted(docs: Mapping[str, list[Union[ALL_DOCUMENTS]]], **numbers: int):
````

Reply:
Yes, but I think that should be a future PR...

Reply:
changed it back and checked it does not affect anything else.